types/ckeditor__ckeditor5-core/ckeditor__ckeditor5-core-tests.ts

```diff
@@ -54,6 +54,10 @@ MyEditor.defaultConfig = { foo: 5 };
  */
 
 class MyPlugin extends Plugin {
+    get pluginName() {
+        return 'MyPlugin';
+    }
```

Comment:
Relying in a string can lead to errors. As I mentioned in the comments, you should check if your variable is `instanceof` a specific `Plugin` subclass.

Reply:
I entirely agree that using the form of `get()` that accepts a string is inherently type-unsafe, and will always need to be coupled with a type casting, or an `instanceof` check (which I also agree is safer than just type casting).

Using the form of `get()` that accepts a plugin class (constructor) reliably returns an instance of that class, so no `instanceof` check or casting is needed in that case, which is the main use case this PR is enabling.

Reply:
I do think you were squashing some problems with the types by using `instanceof`. That's a strong enough check that the typescript compiler will let you treat the value as an instance of the plugin inside the if block, but the type is still incorrect -- e.g. try actually casting the return value to an instance of your plugin class. When I tried the compiler complained that `MyPlugin` and `() => Plugin` don't overlap sufficiently.